### PR TITLE
feat: add dynamic qr code

### LINKED ISSUE #56: Implement Dynamic Mpesa QR API

### PR DESCRIPTION
**Pull request by itsyaasir**
Closes #56

### CODE REVIEW

File: src/services/dynamic_qr.rs [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    /// Name of the Company/M-Pesa Merchant Name
+    merchant_name: &'mpesa str,
+    /// Transaction Reference Number
+    ref_no: &'mpesa str,
+    /// The total amount of the transaction
+    amount: f64,
+    #[serde(rename = "TrxCode")]
+    /// Transaction Type
+    transaction_type: TransactionType,
+
```
Comment (collinsmuriuki) [inline src/services/dynamic_qr.rs:23]:
```suggestion
```

File: src/services/dynamic_qr.rs [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+            credit_party_identifier: express.credit_party_identifier,
+            size: express.size,
+        }
+    }
+}
+
+impl<'mpesa, Env: ApiEnvironment> DynamicQR<'mpesa, Env> {
+    pub(crate) fn builder(client: &'mpesa Mpesa<Env>) -> DynamicQRBuilder<'mpesa, Env> {
+        DynamicQRBuilder::default().client(client)
+    }
```
Comment (collinsmuriuki) [inline src/services/dynamic_qr.rs:98]: Great 👍🏽

**Review by collinsmuriuki — APPROVED** [review]
You implemented the builder well, we should now migrate the existing services to

cc @crispinkoech might need your help here too on this.

**Comment by itsyaasir** [comment]
@collinsmuriuki We can break down the tasks into smaller sub-tasks for the migration to the builder pattern.

@crispinkoech Can look into migrating the bill manager service

**Comment by collinsmuriuki** [comment]
Sounds good @itsyaasir, I can migrate the other apis then, creating issues for this.

**Comment by crispinkoech** [comment]
I can take up the bill manager service.

**Pull request merged**